### PR TITLE
Update extensions docs

### LINKED ISSUE #1491: Extensions design inconsistencies
Different extensions have different assumptions regarding their further extensibility. We need to provide some guidelines and optionally align existing schema files.

**EDIT:** table updated to reflect merged PRs.

| Object | Additional properties | `extras`/`extensions` | `name` |
|-------------------------------------------------------------------|-----------------------|-----------------------|---------|
| `KHR_draco_mesh_compression` | No | No | No |
| `KHR_lights_punctual` root | ✔️ | ✔️ | No |
| `KHR_lights_punctual` `light` | ✔️ | ✔️ | ✔️ |
| `KHR_lights_punctual` `light/spot` | ✔️ | ✔️ | No |
| `KHR_lights_punctual` `node` | ✔️ | ✔️ | No |
| `KHR_materials_pbrSpecularGlossiness` `material` | ✔️ | ✔️ | No |
| `KHR_materials_unlit` `material` | ✔️ | ✔️ | No |
| `KHR_techniques_webgl` root | ✔️ | ✔️ | No |
| `KHR_techniques_webgl` `program`, `shader`, `technique` | ✔️ | ✔️ | ✔️ |
| `KHR_techniques_webgl` `material` | ✔️ | ✔️ | No |
| `KHR_techniques_webgl` `technique/attribute`, `technique/uniform` | ✔️ | ✔️ | No |
| `KHR_texture_transform` | ✔️ | ✔️ | No |
| `EXT_lights_image_based` root | ✔️ | ✔️ | No |
| `EXT_lights_image_based` `light` | ✔️ | ✔️ | ✔️ |
| `EXT_lights_image_based` `scene` | ✔️ | ✔️ | No |

The core spec design is:
- root level arrays elements inherit from `glTFChildOfRootProperty`, thus having `extras`, `extensions`, and `name` and allowing additional properties; [1]
- internal objects with defined fields inherit from `glTFProperty`, thus having `extras`, `extensions` and allowing additional properties. [2]

Current state of extension objects prevent implementations from re-using already existing inheritance chains leading to more potential ecosystem fragmentation.

**Proposal**
Reduce the number of configurations in the table above by making following changes (hoping it's not too late):

- `KHR_lights_punctual`
  - [x] Align root object, `node` and `light/spot` with [2]
- `EXT_lights_image_based`
  - [x] Align `light` with [1]
  - [x] Align root object and `scene` with [2]
- `KHR_texture_transform`
  - [x] Align with [2]

- [ ] Set up a checklist for future extensions proposals that includes inheritance design.

/cc: @pjcozzi @bghgary @emackey @MiiBond @donmccurdy

### PR DESCRIPTION
**Pull request by lexaknyazev**
Fixes #1491.

- Updated Draco extension schema.
- Updated extension examples and guidelines.

### CODE REVIEW

File: extensions/README.md [hunk 1]
```diff
@@ -102,6 +105,8 @@ This allows an engine to quickly determine if it supports the extensions needed
 
 To create a new extension, use the [extension template](Template.md) and open a pull request into this repo.  Make sure to add the extension to the glTF Extension Registry (top of this file).
 
+If the extension adds a new top-level array, its elements should inherit all properties of `glTFChildOfRootProperty.schema.json`. Other objects introduced by the extension should inherit all properties of `glTFProperty.schema.json`. By glTF 2.0 conventions, schemas should allow additional properties. See [`KHR_lights_punctual`](2.0/Khronos/KHR_lights_punctual) as an example.
```
Comment (donmccurdy) [inline extensions/README.md]: By "a new top-level array" I assume we mean an array under the root `extensions` object? Such as:

```
{
  "extensions": {
    "KHR_lights_punctual": {
      "lights": [ ... ]
    }
  }
}
```

I guess an extension could introduce a new *actual* top-level array as well, but currently we haven't done that and I'm not sure we want to. Either way we may want to specify here.

Comment (lexaknyazev) [reply extensions/README.md]: > an array under the root `extensions` object

Yes. _Actual_ new properties could be introduced only by new spec versions.

File: extensions/README.md [hunk 2]
```diff
@@ -136,14 +141,10 @@ In addition to extensions, the `extras` object can also be used to extend glTF.
...
-  "shaders": {
-    "a-vertex-shader-id": {
-      "name": "user-defined-name",
-      "uri" : "vertex-shader.glsl",
-      "type": 35633,
-      "extras" : {
-        "precompiled_binary" : "path_to_precompiled_shader"
-      }
+  "asset": {
+    "verion": 2.0,
```
Comment (bghgary) [inline extensions/README.md]:
```suggestion
    "version": 2.0,
```

File: extensions/README.md [hunk 3]
```diff
@@ -67,32 +67,35 @@ All glTF object properties (see [glTFProperty.schema.json](../specification/2.0/
...
       }
     }
-  }
+  }]
 }
 ```
 * **New parameter semantics**: `CESIUM_RTC` introduces the `CESIUM_RTC_MODELVIEW` semantic.
-* **Reserved IDs**: `KHR_binary_glTF` introduces an explicitly named `buffer` called `binary_glTF`.
+* **Reserved IDs (glTF 1.0 only)**: `KHR_binary_glTF` introduces an explicitly named `buffer` called `binary_glTF`.
 * **New container formats**: `KHR_binary_glTF` introduces a binary format that contains the glTF JSON and geometry, textures, etc.
```
Comment (donmccurdy) [inline extensions/README.md]: The 2nd and 3rd bullets both appear to be 1.0-only. And is `CESIUM_RTC` a current extension? All of this seems outdated. We could remove these three bullets and replace them with more general wording, like:

> Extensions may add new properties and values, such as attribute semantics or texture mime types. In all Khronos (KHR) extensions, and as best practice for vendor extensions, these feature additions are designed to allow safe fallback consumption in tools that do not recognize an extension in the `extensionsUsed` array.

**Pull request merged**